reactor-core/src/main/java/reactor/core/publisher/FluxReplay.java

```diff
@@ -148,9 +159,10 @@
 				long maxAge,
 				Scheduler scheduler) {
 			this.limit = limit;
+			this.indexUpdateLimit = Operators.unboundedOrLimit(limit);
```

Comment:
as demonstrated by (currently failing) `reactor.core.publisher.SinksTest.MulticastReplayN#checkSemanticsSize0`, `limit` can be zero here which would lead to `ArithmeticException`.

Reply:
Remove tests that check zero history size and added protection against passing zero as a history size since it is meaning-less for replay operator (in such scenario we can suggest alternative represented in the `.publish` operator)